zcash_transparent/src/bundle.rs

```diff
@@ -188,11 +182,11 @@ impl OutPoint {
 
     /// Returns the byte representation of the txid of the transaction containing this `OutPoint`.
     pub fn hash(&self) -> &[u8; 32] {
-        self.hash.as_ref()
+        &self.hash
     }
 
     /// Returns the txid of the transaction containing this `OutPoint`.
-    pub fn txid(&self) -> &TxId {
+    pub fn txid(&self) -> &[u8; 32] {
```

Comment:
I think this is the only change to the public API in this PR. A "txid" doesn't make sense in a protocol-specific crate, so I didn't want to move `TxId` into `zcash_transparent`. We could alter `OutPoint` to take a generic parameter, but that would result in _more_ changes to the public API.

Reply:
I have an alternative suggestion: pull `TxId` up into `zcash_protocol`.